### PR TITLE
Migrate from isort/Black/Flake8 to Ruff for Code Formatting and Linting

### PR DESCRIPTION
**Pull request by snejus**
#### Context
I noticed one of recently merged PRs used `unittest` assertions that we are migrating from. Thus, I thought it would be a good idea to configure our linting to catch such issues in the future. Since I was here, I replaced our multiple linting and formatting tools with a single tool, Ruff, to simplify and speed up our development workflow.

#### Summary
This PR migrates the codebase from using isort, Black and Flake8 to Ruff for code formatting and linting. The changes include updates to the GitHub Actions workflow, pre-commit configuration, and documentation to reflect the new tooling.

#### Changes
1. **GitHub Actions Workflow**
   - Updated `.github/workflows/lint.yml` to use `poetry install --only=lint` and `poe lint --output-format=github`.

2. **Pre-commit Configuration**
   - Replaced Black and Isort with Ruff in `.pre-commit-config.yaml`.

3. **Documentation**
   - Updated `CONTRIBUTING.rst` to reflect the use of Ruff for formatting and linting.
   - Modified instructions for running tests and handling external API requests.

4. **Poetry Configuration**
   - Removed Black, Isort, Flake8, and related dependencies from `poetry.lock` and `pyproject.toml`.
   - Added Ruff as the new linter and formatter in `pyproject.toml`.

5. **Setup Configuration**
   - Removed Flake8 configuration from `setup.cfg`.

6. **Git blame**
   - Introduced `.git-blame-ignore-revs` file to keep git blame clean from formatting
   changes. Configure your local `beets` repository to use this file by running:
   ```fish
   $ git config --local blame.ignoreRevsFile .git-blame-ignore-revs
   ```

#### Benefits
- **Performance**: Ruff is known for its speed and efficiency, which should improve the developer experience.
- **Consolidation**: Using a single tool for both formatting and linting simplifies the development workflow.

#### How to Test
1. **Linting and Formatting**
   - Run `poe check-format` to check for formatting issues.
   - Run `poe format` to format the codebase.
   - Run `poe lint` to check for linting issues.

2. **Pre-commit Hooks**
   - Ensure pre-commit hooks are working correctly by running `pre-commit run --all-files`.

3. **CI Pipeline**
   - Verify that the GitHub Actions workflow completes successfully.

#### Notes
- Contributions migrating existing tests from `unittest` to `pytest` are welcome.
- External API requests should be mocked using `requests_mock` and tested weekly in the integration test suite.

#### References
- [Ruff Documentation](https://docs.astral.sh/ruff/)
- [Pre-commit Hooks](https://pre-commit.com/hooks.html)

---

This PR aims to streamline our development process by adopting Ruff, a modern and efficient tool for Python code formatting and linting.

### CODE REVIEW

File: beetsplug/deezer.py
Comment (bal-e) [inline beetsplug/deezer.py:115]: The `!r` component is new, is it intentional?

**Review by bal-e — APPROVED** [review]
For completeness, I've reviewed every file by hand.  In almost all cases, I significantly prefer the changes Ruff has introduced.  I'm happy with this PR overall -- well done @snejus!

File: beetsplug/deezer.py
Comment (snejus) [reply beetsplug/deezer.py:115]: Yep - you can see this replaced the single quotes in the message as `__repr__` for strings adds them.

**Comment by snejus** [comment]
Since this PR depends on `integration_test` marker introduced in #5406 , will wait until its merged before merging this one.

**Comment by snejus** [comment]
> Since this PR depends on `integration_test` marker introduced in #5406 , will wait until its merged before merging this one.

Above does not apply anymore - I rebased it on `master`.

**Review by bal-e — APPROVED** [review]

**Comment by bal-e** [comment]
LGTM, @snejus!

**Comment by snejus** [comment]
Great, I have just updated the `.git-blame-ignore-revs` with updated commit hashes.

**Pull request merged**
